docs/plugins/main.py

```diff
@@ -51,6 +59,8 @@ def on_page_markdown(markdown: str, page: Page, config: Config, files: Files) ->
         return md
     if md := render_why(markdown, page):
         return md
+    if md := render_pydantic_settings(markdown, page):
+        return md
```

Comment:
Ideally we should move away from these pattern: it isn't really elegant and we could move to the [Blocks API](https://facelessuser.github.io/pymdown-extensions/extensions/blocks/) at some point. It seems to provide similar functionality as Sphinx directives which are essential imo